components/UploadForm.vue

```diff
@@ -404,17 +404,22 @@ const submitToArweave = async (record: any): Promise<void> => {
   const arrayBuffer = await record.fileBlob.arrayBuffer()
   let buffer = Buffer.from(arrayBuffer)
   let { ipfsHash } = record
-  if (
-    (record.fileType === 'application/epub+zip' ||
+  if (!record.cachedIpfsHash) {
+    if (
+      (record.fileType === 'application/epub+zip' ||
       record.fileType === 'application/pdf') &&
     isEncryptEBookData.value
-  ) {
-    const { rawEncryptedKeyAsBase64, combinedArrayBuffer } =
+    ) {
+      const { rawEncryptedKeyAsBase64, combinedArrayBuffer } =
       await encryptDataWithAES({ data: arrayBuffer })
-    buffer = Buffer.from(combinedArrayBuffer)
-    key = rawEncryptedKeyAsBase64
-    ipfsHash = await calculateIPFSHash(buffer)
+      buffer = Buffer.from(combinedArrayBuffer)
+      key = rawEncryptedKeyAsBase64
+      record.cachedIpfsHash = await calculateIPFSHash(buffer)
+    } else {
+      record.cachedIpfsHash = record.ipfsHash
+    }
   }
+  ipfsHash = record.cachedIpfsHash
```

Comment:
So when does `encryptDataWithAES ` is run if `isEncryptEBookData ` but `cachedIpfsHash` is set previously?

Reply:
[🎨 Cache IPFS hash in UploadForm](https://github.com/likecoin/nft-book-press/pull/331/commits/e3ff80331b8e5e37b9b814767e8708947a836390)